### PR TITLE
auth: force 'same-site' cookie option

### LINKED ISSUE #507: Fix cookie problem on cross site
As discussed, investigate the fact the auth cookie is not taken into consideration when authenticating (even temp user) using cookies in a cross-domain scenario.

### PR DESCRIPTION
**Pull request by ThomasJuster**
Fixes #507 … by not fixing it actually.

### CODE REVIEW

File: src/app/shared/http-services/auth.http-service.ts [hunk 1]
```diff
@@ -23,7 +23,14 @@ export class AuthHttpService {
...
     use_cookie: '1',
     cookie_secure: appConfig.secure ? '1' : '0',
-    cookie_same_site: appConfig.sameSite ? '1' : '0',
+    /**
+     * NOTE on cookie_same_site:
+     * The 'same-site' policy only applies for auth by third-party cookie.
+     * Safari and Firefox both disable third-party cookies by default, and Chrome will in 2021 (but postponed for 2023)
+     * Since third-party cookies will be removed in a near future, we chose not to enable this type of authentication,
+     * making the 'same-site' policy always true.
+     */
```
Comment (smadbe) [inline src/app/shared/http-services/auth.http-service.ts:32]: 👍

File: src/app/shared/helpers/config.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { environment } from 'src/environments/environment';
 
 interface TokenConfig { authType: 'tokens' }
-type CookieConfig = { authType: 'cookies' } & ({ secure: true, sameSite: boolean } | { secure: boolean, sameSite: true });
+interface CookieConfig { authType: 'cookies', secure: boolean }
```
Comment (smadbe) [inline src/app/shared/helpers/config.ts:4]: Hum... this `secure` is just whether the backend is in https or not. 
We should probably move `authType` to `Environment` directly and compute `cookie_secure` in code.

Can be done in another PR.

**Review by smadbe — APPROVED** [review]
👍 
Small suggestion of improvement but I approve this PR so that you can do it in another PR if you prefer.

**Comment by ThomasJuster** [comment]
Close in favor of #531

**Pull request closed without merging**